scripts.js

```diff
@@ -266,19 +258,23 @@ function handleDropdownRegion() {
     fedsFooter.parentElement.insertBefore(regionDropdownContainer, fedsFooter);
   }
   
-  // Get currently selected region if in the same region page based on localeHome value
-  const currentRegion = regionsNameList.filter(r => r.localeHome === location.pathname);
-  if (!currentRegion.length) {
-    // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use language
-    sessionStorage.getItem('blog-selected-language') || window.blog.language;
-  } else {
-      // uk and apac, just save the info in the session storage
-      sessionStorage.setItem('blog-selected-language', currentRegion[0].lang);
-  }
+  // get actual selected Region
+  getActualRegion = function() {
+    const currentRegion = regionsNameList.filter(r => r.localeHome === location.pathname);
+    let selectedLocale;
+    if (!currentRegion.length) {
+      // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use language
+      selectedLocale = sessionStorage.getItem('blog-selected-language') || window.blog.language;
+    } else {
+      selectedLocale = currentRegion[0].lang;
+        // uk and apac, just save the info in the session storage
+        sessionStorage.setItem('blog-selected-language', currentRegion[0].lang);
+    } return { selectedLocale };
+  } ();
 
   // Automatically build the dropdown based on Locale List
   const dropdownRegionList = document.querySelector('.region-dropdown-list');
-  const selectedRegionLocale = sessionStorage.getItem('blog-selected-language') || window.blog.language;
+  const selectedRegionLocale =  getActualRegion.selectedLocale;
```

Comment:
I find it confusing to call a non-function variable `getSomething`... why use such a complicated way of defining the locale in the first place, with an IIFE returning a single-property object? Instead, I'd return the locale directly and call it like this:
```js
const selectedRegionLocale = getSelectedRegionLocale();
```
Otoh, if you really need to prevent this function from being exposed in the browser and theoretically getting called again (I see no harm in that), I'd simply rename it to `setRegionLocale()` and move the assignment of `selectedRegionLocale` either inside it, or like this:
```js
const selectedRegionLocale = (() => {
  ...
  return selectedLocale;
})();
```

Reply:
Thanks @rofe 

I did change to 

`  // get actual selected Region
  const selectedRegion = () => {
......
    } return {selectedLocale, selectedLocaleName};
  }`

in this case I'm returning not only the selectedLocale, but also the Name as we need it to show in our region Button, like:
![Screen Shot 2021-02-01 at 11 01 17 PM](https://user-images.githubusercontent.com/62952234/106527780-1c01c380-64e8-11eb-94fe-9dd3674b88b1.png)
![Screen Shot 2021-02-01 at 10 31 48 PM](https://user-images.githubusercontent.com/62952234/106527801-2459fe80-64e8-11eb-9491-638e1d8723f1.png)

and I'm calling:
`const {selectedLocale, selectedLocaleName} = selectedRegion(); `

Reply:
Ok, in this constellation I would prefer calling it `getSelectedRegion()` because now it's really a getter 😄

Reply:
(╯°□°)╯︵ ┻━┻ (table flip). 

 np, I can change it so it stays clear.